src/twisted/python/test/test_systemd.py

```diff
@@ -7,158 +7,163 @@
+            of descriptors represented will equal the length of this list.
+        """
+        numFDs = len(names)
+        descriptors = list(range(ListenFDs._START, ListenFDs._START + numFDs))
+        fds = ListenFDs.fromEnvironment(
+            {
+                "LISTEN_PID": str(os.getpid()),
+                "LISTEN_FDS": str(numFDs),
+                "LISTEN_FDNAMES": ":".join(names),
+            }
+        )
+        assert_that(fds, equal_to(ListenFDs(descriptors, tuple(names))))
 
-    def getDaemon(self, start, count):
+    def test_defaultEnviron(self) -> None:
         """
-        Create a new L{ListenFDs} instance, initialized with a fake environment
-        dictionary which will be set up as systemd would have set it up if
-        C{count} descriptors were being inherited.  The descriptors will also
-        start at C{start}.
+        If the process environment is not explicitly passed to
+        L{ListenFDs.fromEnvironment}, the real process environment dictionary
+        is used.
         """
-        fakeEnvironment = self.initializeEnvironment(count, os.getpid())
-        return ListenFDs.fromEnvironment(environ=fakeEnvironment, start=start)
-
-
-class MemoryOnlyTests(MemoryOnlyMixin, InheritedDescriptorsMixin, TestCase):
-    """
-    Apply tests to L{ListenFDs}, explicitly constructed with some fake file
-    descriptors.
-    """
-
-
-class EnvironmentTests(EnvironmentMixin, InheritedDescriptorsMixin, TestCase):
-    """
-    Apply tests to L{ListenFDs}, constructed based on an environment dictionary.
-    """
+        self.patch(os, "environ", initializeEnvironment(5, os.getpid()))
+        sddaemon = ListenFDs.fromEnvironment()
+        self.assertEqual(list(range(3, 3 + 5)), sddaemon.inheritedDescriptors())
 
-    def test_secondEnvironment(self):
+    def test_secondEnvironment(self) -> None:
         """
-        Only a single L{Environment} can extract inherited file descriptors.
+        L{ListenFDs.fromEnvironment} removes information about the inherited
+        file descriptors from the environment mapping.
         """
-        fakeEnvironment = self.initializeEnvironment(3, os.getpid())
-        first = ListenFDs.fromEnvironment(environ=fakeEnvironment)
-        second = ListenFDs.fromEnvironment(environ=fakeEnvironment)
+        env = initializeEnvironment(3, os.getpid())
+        first = ListenFDs.fromEnvironment(environ=env)
+        second = ListenFDs.fromEnvironment(environ=env)
         self.assertEqual(list(range(3, 6)), first.inheritedDescriptors())
         self.assertEqual([], second.inheritedDescriptors())
 
-    def test_mismatchedPID(self):
+    def test_mismatchedPID(self) -> None:
         """
-        If the current process PID does not match the PID in the environment, no
-        inherited descriptors are reported.
+        If the current process PID does not match the PID in the environment,
+        no inherited descriptors are reported.
```

Comment:
Just some extra info about real world use case of this functionality/feature

```suggestion
        If the current process PID does not match the PID in the environment,
        no inherited descriptors are reported.
        
        This is a sign that the systemd sockets are not for us.
        Most probably someone called our code without priour clean of the environment.
```